src/org/exist/xquery/value/BinaryValueFromFile.java

```diff
@@ -82,6 +92,7 @@ public boolean isClosed() {
     @Override
     public void close() throws IOException {
         channel.close();
+        closeListener.ifPresent(cl -> cl.accept(file));
```

Comment:
if ` channel.close();` throws an exception... should we try to execute line 95 as well?

Reply:
Good idea. I will fix...

Reply:
Fixed.